### PR TITLE
Support alias name to be same as selection

### PR DESCRIPTION
**Pull request by fx19880617**
## Description
A fix to support query like:
```
SELECT c1 AS c1 FROM myTable
```

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java [hunk 1]
```diff
@@ -113,32 +113,11 @@ private static void validateSelectionClause(Map<Identifier, Expression> aliasMap
     // Sanity check on selection expression shouldn't use alias reference.
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java:113]: We should not remove the sanity check on usage of alias reference
E.g. we should reject the following query:
```
SELECT colA AS a, foo(a, colB) FROM table
```

Another way is to replace the alias reference in selection as well

Comment (fx19880617) [reply pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java:113]: Added it back, basically remove colA in aliasMap for case like `SELECT colA AS colA`.

Comment (Jackie-Jiang) [reply pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java:113]: I think the correct fix should be just not adding it to the aliasMap?
You can rewrite the query before handling the alias by replacing `a AS a` to `a`

Comment (fx19880617) [reply pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java:113]: done.

File: pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java [hunk 2]
```diff
@@ -113,7 +113,11 @@ private static void validateSelectionClause(Map<Identifier, Expression> aliasMap
...
     Set<String> aliasKeys = new HashSet<>();
     for (Identifier identifier : aliasMap.keySet()) {
-      aliasKeys.add(identifier.getName().toLowerCase());
+      String aliasName = identifier.getName().toLowerCase();
+      if (aliasKeys.contains(aliasName)) {
+        throw new SqlCompilationException("Duplicated alias name found.");
+      }
+      aliasKeys.add(aliasName);
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java:120]: (nit)
```suggestion
      if (!aliasKeys.add(aliasName)) {
        throw new SqlCompilationException("Duplicated alias name found.");
      }
```

**Pull request merged**